lib/src/tag_detection.dart

```diff
@@ -824,125 +830,154 @@ class Tagger {
     bool trustDeclarations = true,
   }) {
-    if (_isBinaryOnly) {
-      Explanation('Binary only',
-          'Cannot assign flutter platform tags, it is a binary only package');
-    } else if (_topLibraries.isEmpty) {
-      Explanation('No top-level libraries found',
-          'Cannot assign Flutter platform tags, because no .dart files were found in lib/');
-    } else {
-      for (final flutterPlatform in _FlutterPlatform.recognizedPlatforms) {
-        final libraryGraph =
-            LibraryGraph(_session, flutterPlatform.runtime.declaredVariables);
-        final declaredPlatformDetector =
-            _DeclaredFlutterPlatformDetector(_pubspecCache);
-        final violationFinder = _PlatformViolationFinder(
-            flutterPlatform,
-            libraryGraph,
-            declaredPlatformDetector,
-            _pubspecCache,
-            runtimeViolationFinder(
-                libraryGraph,
-                flutterPlatform.runtime,
-                (List<Uri> path) => Explanation(
-                    'Package not compatible with runtime ${flutterPlatform.runtime.name} on ${flutterPlatform.name}',
-                    'Because of ${LibraryGraph.formatPath(path)}')));
-
-        // Wanting to trust the plugins annotations when assigning tags we make
-        // a library graph that treats all libraries in plugins as leaf-nodes.
-        //
-        // In this way the plugin restrictions of its dependencies are not
-        // restricting the result.
-        //
-        // We still keep the unpruned detection for providing Explanations.
-        final prunedLibraryGraph = trustDeclarations
-            ? LibraryGraph(_session, flutterPlatform.runtime.declaredVariables,
-                isLeaf: declaredPlatformDetector._isFlutterPlugin)
-            : libraryGraph;
-
-        final prunedViolationFinder = _PlatformViolationFinder(
-            flutterPlatform,
-            prunedLibraryGraph,
-            declaredPlatformDetector,
-            _pubspecCache,
-            runtimeViolationFinder(
-                prunedLibraryGraph,
-                flutterPlatform.runtime,
-                (List<Uri> path) => Explanation(
-                    'Package not compatible with runtime ${flutterPlatform.runtime.name} of ${flutterPlatform.name}',
-                    'Because of ${LibraryGraph.formatPath(path)}')));
-        // Report only the first non-pruned violation as Explanation
-        final firstNonPrunedViolation = _topLibraries
-            .map(violationFinder._findPlatformViolation)
-            .firstWhere((e) => e != null, orElse: () => null);
-        if (firstNonPrunedViolation != null) {
-          explanations.add(firstNonPrunedViolation);
-        }
+    try {
+      if (_isBinaryOnly) {
+        Explanation('Binary only',
+            'Cannot assign flutter platform tags, it is a binary only package');
+      } else if (_topLibraries.isEmpty) {
+        Explanation('No top-level libraries found',
```

Comment:
shouldn't this be `explanations.add(...)`?

Reply:
yes!!